### ISSUE TITLE
Port IMPROVER to Python 3

### ISSUE THREAD

**Issue by arh89** [metoppv/improver#488]
As an IMPROVER developer, I want to port our code to Python 3 in order to avoid problems when Python 2 is no longer supported in 2020.

Related issues: Scoping ticket: #487 and epic: #265 

Acceptance criteria:
 * Following the suggestions in ticket #487, port the remaining code to Python 3.
 * Alter the Travis configuration to ensure that all new code/PRs are Python 3 compatible (do not merge into master until tests pass).

**Comment by gavinevans** [comment]
Ported to Python 3 in #558.
